pkgs/development/python-modules/python-arango/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, mock
+, pyjwt
+, pytest
+, requests
+, requests-toolbelt
+}:
+
+buildPythonPackage rec {
+  pname = "python-arango";
+  version = "7.5.0";
+  disabled = pythonOlder "3.7";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "ArangoDB-Community";
+    repo = "python-arango";
+    rev = version;
+    sha256 = "0mpljnry5av0r0b0dvsqylpih6v6hkhbz1pbsdp3a8kjlybfl2ya";
+  };
+
+  pythonImportsCheck = [ "arango" ];
+
+  nativeBuildInputs = [
+    mock
+    pytest
+  ];
```

Comment:
For this one, Nix offers `pytestCheckHook`. You can import it and use on `checkInputs` instead. If there are tests that require internet connection (I saw some from this library), you can use `disableTestPaths` to disable whole test files or `disableTests` for single tests. You can use `python-modules/aioridwell` as an example.

Reply:
Thank you for the pointer! I am not a python expert, so that was very helpful. Patches with running tests underway...

Reply:
Ok I tried, but the tests all require a running `arangodb` instance which is non-trivial.

Reply:
One option is that if all tests require `arangodb` to be working, then you could disable most of them or set `doCheck = false` if they all depend of it (adding the proper comment of course). If you use the latter, then you don't need to import `pytestCheckHook`.

Reply:
Update: figured out how to spawn a test db. Using arangodb 3.10.0 from https://github.com/NixOS/nixpkgs/pull/194670 - tests run now. Though there are some failing tests I have disabled.

Some disabled tests seem unlikely to be safe to disable.

Reply:
For reference: I disabled 27 disabled tests of ~150 total. So this is still likely reasonable, but I am still a bit suspicious. Should I investigate further? `python-arango` 7.5.1 seems to have corrected for many changes from `arangodb` 3.10.0. My gut feeling is that the test failures are mostly quirks to do with running in the sandbox.

Reply:
As long as you can justify the reason these tests were disabled, then all good. There's not much to be done if the issue is the sandbox.